### ISSUE TITLE
[dotnet 6] `dotnet test` does not allow passing `--arch` and `--results-directory` together

### ISSUE THREAD

**Issue by miarodri** [dotnet/sdk#24728]
### Describe the bug
When trying run `dotnet test` with both the `--arch` and `--results-directory`, the program fails with the error:

```txt
Specifying both the `-r|--runtime` and `-a|--arch` options is not supported.
```

I only have the vaguest of ideas why this is, but I assume it has to do with the short argument flag for `--results-directory` being `-r` which clashes with the short arg for `--runtime`. Looking at the code, it seems like this may be fixed for dotnet 7? [At the very least it seems like the `-r` option for `--results-directory` was removed](https://github.com/dotnet/sdk/commit/59485cceb1d439f24b4c418ba0aad9e8c38f47c7) and the result command was modified [when removing obsolete APIs](https://github.com/dotnet/sdk/commit/a086910105c2ddda02e2d46ee12b005cdb98bc4b).

### To Reproduce
```pwsh
dotnet test --arch x64 --results-directory ./results any.csproj
```

### Further technical details
```pwsh
PS > dotnet --info
.NET SDK (reflecting any global.json):
 Version:   6.0.201
 Commit:    ef40e6aa06

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.22000
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\6.0.201\

Host (useful for support):
  Version: 6.0.3
  Commit:  c24d9a9c91

.NET SDKs installed:
  3.1.417 [C:\Program Files\dotnet\sdk]
  5.0.212 [C:\Program Files\dotnet\sdk]
  5.0.406 [C:\Program Files\dotnet\sdk]
  6.0.201 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.All 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.23 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.15 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.3 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.30 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.23 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.23 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.3 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
```

Running on Powershell Core:
```pwsh
PS > $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.2.2
PSEdition                      Core
GitCommitId                    7.2.2
OS                             Microsoft Windows 10.0.22000
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by nohwnd** [comment]
@MarcoRossignoli please take a look on this. If it really fails like that using the full names of the parameters there is maybe some additional translation to `-r` and mixup of what it means. At least that is my guess.

**Comment by MarcoRossignoli** [comment]
@miarodri the issue was here https://github.com/dotnet/sdk/blob/release/6.0.2xx/src/Cli/dotnet/CommonOptions.cs#L145 until 6.0.20x and it was been fixed here https://github.com/dotnet/sdk/blob/release/6.0.3xx/src/Cli/dotnet/CommonOptions.cs#L177 in 6.0.300 

Can you upgrade the .NET SDK to the 6.0.300 shipped with 6.0.5? You can download from here https://dotnet.microsoft.com/en-us/download/dotnet/6.0
